maplap/detector.py

```diff
@@ -0,0 +1,222 @@
+                                       adaptiveMethod=cv2.ADAPTIVE_THRESH_GAUSSIAN_C,
+                                       thresholdType=cv2.THRESH_BINARY_INV,
+                                       blockSize=block_size,  # Parameter!
+                                       C=level_black)  # Parameter!
+    return gray_image
+
+
+def detect_lines_without_width(const_image: np.ndarray,
+                               threshold_line: int = 100,
+                               min_line_length: int = 15,
+                               max_line_gap: int = 10) -> list:
+    coordinates: list = cv2.HoughLinesP(const_image,
+                                        rho=1,
+                                        theta=math.pi / 180,
+                                        threshold=threshold_line,  # threshold for line detection
+                                        minLineLength=min_line_length,  # minimal line length
+                                        maxLineGap=max_line_gap)  # max gap between two lines
+    lines = []
+    if coordinates is not None:
+        for coordinate in coordinates:
+            x_first, y_first, x_second, y_second = coordinate[0]
+            lines.append(Line(x_first, y_first, x_second, y_second))
+    return lines
+
+
+def count_intersections(gray_image: np.ndarray,
+                        x_center: int,
+                        y_center: int,
+                        radius: int) -> int:
+    count: int = 0
+    if x_center - radius < 0 or x_center + radius >= gray_image.shape[1]\
+            or y_center - radius < 0 or y_center + radius >= gray_image.shape[0]:
+        return 0
+    for phi in np.arange(0.0, 2 * np.pi, 5 / radius):
+        if gray_image[round(y_center + radius * math.sin(phi))]\
+           [round(x_center + radius * math.cos(phi))] != 0:
+            count = count + 5
+    return count
+
+
+def detect_centers_of_circles(gray_image: np.ndarray,
+                              find_rate: int = 1,
+                              threshold_center: int = 20,
+                              min_radius: int = 0,
+                              max_radius: int = 0) -> list:
+    centers: list = cv2.HoughCircles(gray_image,
+                                     cv2.HOUGH_GRADIENT,
+                                     find_rate,
+                                     minDist=0.0001,
+                                     param1=50,  # threshold for Canny edge detector
+                                     param2=threshold_center,  # threshold for center detection
+                                     minRadius=min_radius,
+                                     maxRadius=max_radius)
+    if centers is not None:
+        circles: list = []
+        if centers is not None:
+            for center in centers[0, :]:
+                x_center, y_center, radius = map(int, center)
+                circles.append(Circle(x_center, y_center, radius, 1))
+        return circles
+    return list()
```

Comment:
Обычно, вместо вызова конструктора используют специальный литералы:
```suggestion
    return []
```